src/components/molecules/SubNavigation/SubNavigation.jsx

```diff
@@ -1,23 +1,38 @@
 import "./SubNavigation.scss";
 import Arrow from "../../atoms/assets/svg/angle-left-solid-black.svg";
+import React, { useEffect, useRef } from "react";
 
-const SubNavigation = ({ subnav, openSubmenu, backBtnMenu }) => {
-  
+const SubNavigation = ({ subnav, setIsSubMenuOpen, backBtnMenu }) => {
   const reduceItem = subnav.reduce((subMenu, item) => {
     if (item.isSubMenu) return { ...subMenu, item };
     return subMenu;
   }, []);
 
+  const subMenuRef = useRef();
+  useEffect(() => {
+    const handler = (e) => {
+      if (subMenuRef.current && !subMenuRef.current.contains(e.target)) {
+        setIsSubMenuOpen(false);
+      }
+    };
+
+    document.addEventListener("click", handler);
```

Comment:
Don't bind events this way. In react you should bind to components directly

Reply:
Solved